skylark_library.bzl

```diff
@@ -0,0 +1,103 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylib module containing a library rule for aggregating rules files."""
+
+SkylarkLibraryInfo = provider()
+
+def _skylark_library_impl(ctx):
+  all_files = depset(ctx.files.srcs, order="postorder")
+  for dep in ctx.attr.deps:
+    all_files += dep.files
+  return [
+      # All dependent files should be listed in both `files` and in `runfiles`;
+      # this ensures that a `library()` can be referenced as `data` from a
+      # separate program, or from `tools` of a genrule().
+      DefaultInfo(
+          files = all_files,
```

Comment:
Please remove spaces around `=` in kwargs lists (here, below, and in the `rule` call).

Reply:
Done-ish. Didn't quite know what you were referring to with the `rule` call; there's documentation about how to use it in a `BUILD` file which has different style guides.

Reply:
In the call to the `rule` function at the end of the file, where the rule is defined, those are kwargs so I don't put any spaces around the `=`. The only places where I use spaces around `=` are in BUILD files when creating a target based on a rule, or in macros where a rule is being invoked.

It's an unfortunate style inconsistency, I'll admit.

Reply:
Done.